pkg/sql/schemachanger/scplan/internal/rules/current/op_drop.go

```diff
@@ -172,6 +172,7 @@ func init() {
 				(*scpb.CheckConstraint)(nil),
 				(*scpb.ForeignKeyConstraint)(nil),
 				(*scpb.UniqueWithoutIndexConstraint)(nil),
+				(*scpb.ColumnNotNull)(nil),
```

Comment:
Perhaps we should have a filter in the helpers for non-index-backed constraints?